src/graphql/res_cache.rs

```diff
@@ -0,0 +1,116 @@
+use std::sync::{Arc, RwLock};
+use std::time::Duration;
+
+use async_graphql_value::ConstValue;
+use ttl_cache::TtlCache;
+
+use crate::config::Cache;
+use crate::lambda::{EvaluationContext, Expression, ResolverContextLike};
+
+const DEFAULT_MAX_AGE: u64 = 30;
+
+#[derive(Clone)]
+pub struct ResCache {
+  cache: Cache,
+  data: Arc<RwLock<TtlCache<u64, ConstValue>>>,
+}
+
+impl std::fmt::Debug for ResCache {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(
+      f,
+      "ResCache {{cache: {:?}, capacity: {:?}}}",
+      self.cache,
+      self.data.read().unwrap().capacity()
+    )
+  }
+}
+
+impl ResCache {
+  pub fn new(cache: Cache) -> Self {
+    ResCache { cache, data: Arc::new(RwLock::new(TtlCache::new(1000))) }
+  }
+
+  fn insert(&self, key: u64, value: &ConstValue) -> Option<ConstValue> {
```

Comment:
```suggestion
  fn insert(&self, value: &A) -> Option<bool> where A: Hasher {
```
Could this work?

Reply:
made the `value` generic but the return type is same as the type of `value` since the `TtlCache` library returns the previously existing value, so I'm putting the responsibility on the caller to use or ignore the value